src/OpenLoco/Paint/Paint.cpp

```diff
@@ -41,10 +42,128 @@ namespace OpenLoco::Paint
         return &_session;
     }
 
+    void registerHooks()
+    {
+        registerHook(
+            0x004622A2,
+            [](registers& regs) -> uint8_t {
+                registers backup = regs;
+
+                PaintSession session;
+                session.generate();
+
+                regs = backup;
+                return 0;
+            });
+    }
+
+    // 0x00461CF8
+    static void tilePaintSetup(PaintSession& session, const Map::map_pos& loc)
+    {
+        registers regs{};
+        regs.eax = loc.x;
+        regs.ecx = loc.y;
+        call(0x00461CF8, regs);
+    }
+
+    // 0x004617C6
+    static void tilePaintSetup2(PaintSession& session, const Map::map_pos& loc)
```

Comment:
Any idea what's the difference between 1 and 2?

Reply:
Not sure yet for the tiles but for the entities it's for flying/floating entities